### ISSUE TITLE
Strict CI for merging to develop branch

### ISSUE THREAD

**Issue by birchmd** [aurora-is-near/aurora-engine#89]
We want to automate deployment to testnet, however we also do not want to accidentally break anything. Therefore, we need strict CI tests which will ensure merging the new code will not break existing features.

Suggested workflow:
1. Create a `staging.aurora` account on Near
2. Deploy the contract to `staging.aurora` (and initialize the contract)
3. Run through some transactions:
    a. Mint ERC-20 + Transfer ERC-20
    b. Access eth faucet? (TODO: not sure how this will work with bridged eth yet)
    c. Interact via relayer? (TODO: this would require setting up a new relayer for staging, is this worth it?)

This CI suite will likely take a long time to run, so I don't think it should trigger on every push. It also has a stateful impact on testnet, so multiple of these runs cannot be done in parallel. Maybe we should forbid PRs directly targeting `develop` and instead let people merge into another branch (e.g. `staging`) which automatically tries to deploy to testnet once per day, and if it is successful then merges to `develop`?

Suggestions for the transactions performed, and how to setup the triggering of these tests are welcome.

**Comment by birchmd** [comment]
@artob noted that the naming here is a little strange because typically the levels of deployment are dev -> staging -> production (possibly with other intermediate stages).

The reason the names came out the way they did was because the the `develop` branch and `develop.aurora` Near account already exist. And `staging` is a natural name to describe something that being prepared to be pushed out to a more important environment.

I'm open to suggestions for naming.
